ec2utils/ec2uploadimg/lib/ec2utils/ec2uploadimg.py

```diff
@@ -275,18 +275,14 @@ def _create_image_root_volume(self, source):
             self._check_security_groups_exist()
         if self.running_id:
             # When using an already running instance, simply look up this one
-            helper_instance = self._get_helper_instance()
+            self.helper_instance = self._get_helper_instance()
         else:
-            helper_instance = self._launch_helper_instance()
-        self.helper_instance = helper_instance
+            self.helper_instance = self._launch_helper_instance()
```

Comment:
In this case I'd like to keep the current logic, i.e.

if self.running_id:
            # When using an already running instance, simply look up this one
            helper_instance = self._get_helper_instance()
        else:
            helper_instance = self._launch_helper_instance()
        self.helper_instance = helper_instance

Just a style thing, but also I think "self.helper_instance = self._get_helper_instance()" can be confusing to the reader as it would be easy for the reader to assume that self._get_helper_instance() just returns self.helper_instance. Therefore the current proposed construct can easily be confused as being self referential.

Reply:
Okay. I've dropped the first commit.